### ISSUE TITLE
DXE-2675 Third party certificate `sans` cause perpetual in-place update of already created resource

### ISSUE THREAD

**Issue by james-downing** [akamai/terraform-provider-akamai#415]
### Terraform Version
Terraform: v1.2.5
Akamai Provider: v3.6.0

### Affected Resource(s)
- akamai_cps_third_party_enrollment

### Terraform Configuration Files
```hcl
resource "akamai_cps_third_party_enrollment" "enrollment" {
  contract_id = data.akamai_contract.redacted.id
  common_name = var.hostname
  sans = ["terraform-foo.dev.redacted.co.uk"]

  secure_network         = "enhanced-tls"
  signature_algorithm    = "SHA-256"
  sni_only               = true
  certificate_chain_type = "default"
  change_management      = false
  exclude_sans           = false

  admin_contact {
    first_name       = var.contact_first_name
    last_name        = var.contact_second_name
    phone            = var.contact_redacted_phone
    email            = var.contact_email_address
    address_line_one = var.contact_address_line_one
    address_line_two = var.contact_address_line_two
    city             = var.contact_city
    country_code     = var.contact_country_code
    organization     = var.contact_organisation
    postal_code      = var.contact_postal_code
    region           = var.contact_region
    title            = ""
  }

  tech_contact {
    first_name       = "redacted"
    last_name        = ""
    phone            = "redacted"
    email            = "redacted"
    address_line_one = var.contact_address_line_one
    city             = var.contact_city
    country_code     = var.contact_country_code
    organization     = var.organization_name
    postal_code      = var.contact_postal_code
    region           = var.contact_region
    title            = "Administrator"
  }

  csr {
    country_code        = var.contact_country_code
    city                = var.contact_city
    organization        = var.organization_name
    organizational_unit = ""
    state               = var.contact_region
  }

  network_configuration {
    clone_dns_names         = false
    disallowed_tls_versions = ["TLSv1", "TLSv1_1"]
    geography               = "core"
    must_have_ciphers       = var.akamai_ciphers
    ocsp_stapling           = "on"
    preferred_ciphers       = var.akamai_ciphers
    quic_enabled            = false
  }

  organization {
    name             = var.organization_name
    phone            = var.organization_phone
    address_line_one = var.organization_address_line_one
    city             = var.organization_city
    country_code     = var.organization_country_code
    postal_code      = var.organization_postal_code
    region           = var.organization_region
  }
}

data "akamai_contract" "redacted" {
  group_name = var.akamai_contract_group_name
}

```
### Expected Behavior
Once the above resource is applied and resource created, subsequent applications should show no changes.

### Actual Behavior
```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.akamai_certificate_redacted_dev["host_two"].akamai_cps_third_party_enrollment.enrollment will be updated in-place
  ~ resource "akamai_cps_third_party_enrollment" "enrollment" {
        id                                    = "180758"
      ~ sans                                  = [
          - "terraform-foo.dev.redacted.co.uk",
            # (1 unchanged element hidden)
        ]
        # (9 unchanged attributes hidden)





        # (5 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```
This causes the CSR to be recreated. I haven't yet tested the impact of this for a certificate which has the data uploaded via the `akamai_cps_upload_certificate` resource. If it causes a new CSR to be regenerated and invalidating the old cert data that would be extremely problematic.

### Steps to Reproduce
1. `terraform apply` 
2. `terraform apply` again

**Comment by dstopka** [comment]
Hello @james-downing,

Thank you for opening an issue. I'll be working on investigating it and I'll get back to you once I know more about it.

Kind regards,
Darek

**Comment by james-downing** [comment]
Thanks @dstopka 

I was doing some more work around this today and noticed the following pattern, using the above code still as an example:

1. After running the first apply, the certificate configuration will include both the `common_name` and `sans` under the SAN list within the UI.
2. Subsequently, running Terraform plan+apply Terraform wants to remove the `common_name` from the SAN list. (This makes sense in terms of matching the given configuration).
3. Running an apply on this will be seen as successful however no changes are made to the certificate (Both the `common_name` and `sans` remain within the SAN list)
4. Terraform is now stuck in this cycle.

As a workaround I'm including the `common_name` within the `sans` list.

**Comment by lkowalsk-akamai-com** [comment]
This issue was fixed in 5.0 release
